packages/field-plugin/helpers/vite/src/sandbox.ts

```diff
@@ -0,0 +1,64 @@
+import * as querystring from 'querystring'
+import { bold } from './utils/text'
+import { arrows } from './utils/arrows'
+import { load, fileExists, MANIFEST_FILE_NAME } from './manifest'
+import type { Manifest } from './manifest'
+
+export const SANDBOX_BASE_URL = `https://plugin-sandbox.storyblok.com/field-plugin`
+
+export type SandboxQueryParams = {
+  url: string
+  manifest: Manifest | null
+}
+
+export const buildQueryString = (params: SandboxQueryParams) => {
+  const queryParams: any = {
+    url: params.url,
+    manifest: JSON.stringify(params.manifest),
+  }
+
+  return querystring.stringify(queryParams)
+}
+
+export const generateSandboxUrl = (fieldPluginUrl: string) => {
+  displayManifestChecking()
+
+  const queryParams: SandboxQueryParams = {
+    url: fieldPluginUrl,
+    manifest: loadManifest(),
+  }
+
+  const queryString = buildQueryString(queryParams)
+
+  return `${SANDBOX_BASE_URL}?${queryString}`
+}
+
+const loadManifest = (): Manifest | null => {
+  if (!fileExists()) return null
+
+  try {
+    return load()
+  } catch (err) {
+    displayManifestErrorLoading(err as Error)
+    return null
+  }
+}
+
+const displayManifestChecking = () => {
+  if (fileExists()) {
+    displayManifestFileExists()
+  } else {
+    displayManifestFileNotFound()
+  }
+}
+
+const displayManifestFileExists = () =>
+  console.log(
+    `${arrows.green} ${bold(`Loading manifest file: ${MANIFEST_FILE_NAME}`)}`,
+  )
+
+const displayManifestFileNotFound = () =>
+  console.log(`${arrows.gray} ${bold(`Manifest file not found`)}`)
```

Comment:
If we're not meant to reuse these somewhere else, what if we embed them directly within `displayManifestChecking`? We can jump around functions less.

Reply:
thanks for the suggestion @eunjae-lee, I was also feeling it 'too much'.

I had just updated it. 🙌